baseframe/templates/baseframe/bootstrap3/forms.html.jinja2

```diff
@@ -111,13 +111,34 @@
   {% if message %}<p>{{ message }}</p>{% endif %}
   <div style="display:none;"><input type="hidden" name="form.id" value="{{ formid }}" /></div>
   {{ form.hidden_tag() }}
-  {%- set autofocus = true %}
+  <!--  In jinja, global scoped variables can't be modified inside a loop. https://github.com/pallets/jinja/issues/641. Hence using a dictionary to set autofocus and check if it is first field.
+  Autofocus is set true for the first field of the form provided it is not radio or checkbox.
+  In case of validation errors, autofocus is set true for the first field with an error.-->
```

Comment:
Use Jinja2 comments (`{# Comment #}`). HTML comments will be included in the rendered page.

Reply:
Also, this code seems a bit obtuse. Try this:

1. Figure out autofocus field (if any at all) in Python code
2. Pass the field object to `renderform`
3. Let renderform pass `autofocus=True` to `renderfield`